Add Anglian Water documentation

pantherale0 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Add documentation for new Anglian Water core integration.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [X] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [X] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/156225
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/8402
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

pantherale0 committed:
Create Anglian Water integration documentation

Add Anglian Water integration for smart meter data access

c0ffeeca7 reviewed (changes requested):
Please remove comments.

joostlek commented on source/_integrations/anglian_water:
```diff
@@ -0,0 +1,83 @@
+ha_integration_type: integration
+ha_platforms:
+  - sensor
+ha_category: []
```
empty?

pantherale0 replied:
Added Sensor, I wasn't sure if we could add Water Management here or not.

pantherale0 committed:
Revise Anglian Water integration documentation

Updated documentation for Anglian Water integration, including category, device support, configuration details, and troubleshooting information.

c0ffeeca7 commented on source/_integrations/anglian_water:
```diff
@@ -0,0 +1,83 @@
+ha_config_flow: true
+---
+
+**Anglian Water** {% term integration %} is used to integrate with the smart meters of [Anglian Water](https://www.anglianwater.co.uk).
```
```suggestion
The **Anglian Water** {% term integration %} is used to integrate with the smart meters of [Anglian Water](https://www.anglianwater.co.uk).
```

c0ffeeca7 committed:
tiny tweak

c0ffeeca7 reviewed (approved):
Thank you, @pantherale0 :+1:
